common/persistence/cassandraPersistence.go

```diff
@@ -2415,12 +2415,16 @@ func (d *cassandraPersistence) updateSignalInfos(batch *gocql.Batch, signalInfos
 	}
 }
 
-func (d *cassandraPersistence) updateSignalsRequested(batch *gocql.Batch, signalReqIDs []string, deleteSignalReqID string,
+func (d *cassandraPersistence) updateSignalsRequested(batch *gocql.Batch, signalReqIDs map[string]struct{}, deleteSignalReqID string,
 	domainID, workflowID, runID string, condition int64, rangeID int64) {
 
+	signalReqIDsSlice := []string{}
```

Comment:
and this is even stranger where you changed the parameter from slice to a map and then convert it back to a slice within the function.

Reply:
the database layer is using set, and i cannot simply just use a map here.